### PR TITLE
Add xray api with insights to our codebase

### PR DESCRIPTION
**Pull request by zoltanbedi**
## TODO:

- [x] Add test for GetInsight
- [x] Do not show query input for insight type
- [x] Add dropdown for State

### CODE REVIEW

File: pkg/datasource/getInsights.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+	for _, query := range req.Queries {
+		response.Responses[query.RefID] = getSingleInsight(xrayClient, query)
+	}
+
+	return response, nil
+}
+
+func getSingleInsight(xrayClient XrayClient, query backend.DataQuery) backend.DataResponse {
+	insightsResponse, err := xrayClient.GetInsightSummaries(&xray.GetInsightSummariesInput{
```
Comment (aocenas) [inline pkg/datasource/getInsights.go:38]: Isn't there a `GetInsightSummariesPages` api? If so we should probably use that so we can limit the number of results if needed.

Comment (zoltanbedi) [reply pkg/datasource/getInsights.go:57]: There isn't. But there is a GetInsightSummariesWithContext. Should that be used instead?

Comment (aocenas) [reply pkg/datasource/getInsights.go:57]: Probably no need right now. Context makes sense in cases you are doing async stuff with go routines and you can pass different timeouts or link go routines that should be ended together in a group but probably no need for that atm.

File: pkg/datasource/getInsights.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	log.DefaultLogger.Debug("states", "states", states)
+
+	if queryData.State == "All" || len(queryData.State) == 0 {
+		states = nil
+	}
+
+	insightsResponse, err := xrayClient.GetInsightSummaries(&xray.GetInsightSummariesInput{
+		StartTime: &query.TimeRange.From,
+		EndTime:   &query.TimeRange.To,
+		States:    aws.StringSlice(states),
```
Comment (aocenas) [inline pkg/datasource/getInsights.go:60]: What is the difference between String and StringSlice?

File: pkg/datasource/getInsights.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+	if err != nil {
+		return backend.DataResponse{
+			Error: err,
+		}
+	}
+
+	var states []string = []string{strings.ToUpper(queryData.State)}
+
+	log.DefaultLogger.Debug("states", "states", states)
```
Comment (aocenas) [inline pkg/datasource/getInsights.go:51]:
```suggestion
	log.DefaultLogger.Debug("getSingleInsight", "states", states, "group", queryData.Group)
```

File: pkg/datasource/getInsights.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+	for _, insight := range insightsResponse.InsightSummaries {
+		description := strings.Split(aws.StringValue(insight.Summary), ".")[1]
+		description = strings.TrimSpace(description) + "."
+		rootCauseService := fmt.Sprintf("%s (%s)", aws.StringValue(insight.RootCauseServiceId.Name), aws.StringValue(insight.RootCauseServiceId.Type))
+		anomalousService := fmt.Sprintf("%s (%s)", aws.StringValue(insight.TopAnomalousServices[0].ServiceId.Name), aws.StringValue(insight.TopAnomalousServices[0].ServiceId.Type))
+		responseDataFrame.AppendRow(
+			insight.InsightId,
+			description,
+			int64(insight.EndTime.Sub(aws.TimeValue(insight.StartTime))/time.Millisecond),
```
Comment (aocenas) [inline pkg/datasource/getInsights.go:90]: Who does the formatting here? Seems like we show less information but I assume the issue is somewhere in the table component auto formatting or something?
Screenshot from 2020-09-03 11-40-14
Screenshot from 2020-09-03 11-40-09

**Review by aocenas — APPROVED** [review]

File: pkg/datasource/getInsights.go (hunk 2 above)
Comment (zoltanbedi) [reply pkg/datasource/getInsights.go:60]: It is the same as String but for String arrays.

File: pkg/datasource/getInsights.go (hunk 4 above)
Comment (zoltanbedi) [reply pkg/datasource/getInsights.go:90]: I set the format here https://github.com/grafana/x-ray-datasource/pull/28/files#diff-c421e8ad1824f9127d38c4f6f70d5377R107

**Pull request merged**